From 705a29cb71e1fce1c53384dd23550fb566dc7f1b Mon Sep 17 00:00:00 2001
From: Luwei Zhou <b45643@freescale.com>
Date: Wed, 17 Sep 2014 13:03:03 +0800
Subject: [PATCH] ENGR00331509-2 ARM: dts: i.mx6q/dl/sx: Add isl29023 device
 tree support.

Add isl29023 sensor device tree support on i.MX6Q/DL/SX platform.

Signed-off-by: Luwei Zhou <b45643@freescale.com>
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi |  9 +++++++++
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi   | 10 ++++++++++
 arch/arm/boot/dts/imx6sx-sdb.dts         |  9 +++++++++
 3 files changed, 28 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index acc33b0..0f3dca9 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -355,6 +355,14 @@
 		interrupt-parent = <&gpio2>;
 		interrupts = <29 1>;
 	};
+
+	isl29023@44 {
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <17 2>;
+	};
 };
 
 &mlb {
@@ -385,6 +393,7 @@
 				MX6QDL_PAD_SD2_DAT0__GPIO1_IO15  0x80000000
 				MX6QDL_PAD_EIM_BCLK__GPIO6_IO31 0x80000000
 				MX6QDL_PAD_EIM_EB1__GPIO2_IO29	0x80000000
+				MX6QDL_PAD_DISP0_DAT23__GPIO5_IO17 0x80000000
 			>;
 		};
 
diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index 6475c24..a21db98 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -431,6 +431,15 @@
 		interrupt-parent = <&gpio3>;
 		interrupts = <16 1>;
 	};
+
+	isl29023@44 {
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		vdd-supply = <&reg_sensor>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <9 2>;
+	};
 };
 
 &iomuxc {
@@ -459,6 +468,7 @@
 				MX6QDL_PAD_EIM_D16__GPIO3_IO16 0x80000000
 				MX6QDL_PAD_SD3_RST__GPIO7_IO08	0x80000000
 				MX6QDL_PAD_GPIO_9__GPIO1_IO09 	0x80000000
+				MX6QDL_PAD_EIM_DA9__GPIO3_IO09 0x80000000
 			>;
 		};
 
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index d483caf..09ef2b9 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -495,6 +495,15 @@
 		interrupts = <5 1>;
 		shared-interrupt;
 	};
+
+	isl29023@44 {
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <5 1>;
+		shared-interrupt;
+	};
 };
 
 &i2c4 {
-- 
1.8.0

